### PR TITLE
[finishes #101090500] Outline img and labels when selecting and dragg…

### PR DESCRIPTION
**Pull request by mehalshah**
…ing them in applab

This helps users know what they are editing and how big it is - useful for labels and images with white backgrounds

### CODE REVIEW

**Comment by davidsbailey** [comment]
Thanks Mehal, I'll take a look. Could you please add a screenshot or video of the new behavior so I can see how it looks?

File: apps/src/applab/designMode.js [hunk 1]
```diff
@@ -569,11 +572,14 @@ function makeDraggable (jqueryElements) {
 
         elm.css({
           top: newTop,
-          left: newLeft
+          left: newLeft,
```
Comment (davidsbailey) [inline apps/src/applab/designMode.js]: revert stray comma (unless there was supposed to be more css here?)

Comment (mehalshah) [inline apps/src/applab/designMode.js]: Reverted

File: apps/src/applab/designMode.js [hunk 2]
```diff
@@ -650,6 +656,30 @@ function makeUndraggable(jqueryElements) {
...
+      outlineWidth: '1px',
+    });
+  }
+}
+
+/**
+ * Remove dashed borders from all elements
+ */
+function removeElementHighlights() {
+  $('#designModeViz div').parent().css({
```
Comment (davidsbailey) [inline apps/src/applab/designMode.js]: How about `$('desigModeViz .ui-draggable).css({...})` ? Although this change looks ok now, I'm concerned about what will happen if we try to put an outline on a div for a different reason down the road.

Comment (mehalshah) [inline apps/src/applab/designMode.js]: Doing that would apply the dashed lines to all controls, not just labels and imgs. Dashes are unnecessary (and look bad) for most elements.

File: apps/src/applab/designMode.js [hunk 3]
```diff
@@ -650,6 +656,30 @@ function makeUndraggable(jqueryElements) {
   return foundOne;
 }
 
+/**
+ * Highlights an element with a dashed border, removes border from all other elements
```
Comment (davidsbailey) [inline apps/src/applab/designMode.js:662]: How about adding something like "must only be called on an element wrapped in a draggable div" to the comments (or instead checking that it's parent is draggable if you prefer).

File: apps/src/applab/designMode.js (hunk 2 above)
Comment (davidsbailey) [inline apps/src/applab/designMode.js]: wouldn't this remove dashes from all controls, which is what you are doing here already?

**Comment by mehalshah** [comment]
dragging

File: apps/src/applab/designMode.js (hunk 3 above)
Comment (mehalshah) [inline apps/src/applab/designMode.js:662]: Sure

File: apps/src/applab/designMode.js [hunk 4]
```diff
@@ -569,11 +572,14 @@ function makeDraggable (jqueryElements) {
...
+          left: newLeft,
         });
 
         designMode.renderDesignWorkspace(elm[0]);
-      }
+      },
+      start: function () {
+        highlightElement(elm[0]);
+      },
     }).css({
```
Comment (davidsbailey) [inline apps/src/applab/designMode.js:585]: I think we want highlightElement to happen whenever an element is selected, so maybe it should be called from within editElementProperties instead? In any case it seems to me that the outline should appear after I drag an element into the visualization area from the toolbox the same way it would if I drag it around within the visualization area. Let me know if you disagree.

Likewise it seems like it should work if I click an element to select it (but do not drag), though it looks like that might work already.

**Comment by davidsbailey** [comment]
Hey Mehal, nice work, this looks like it works according to spec. However, I have few a paranoid comments about assumptions changing in the future and one comment about the desired behavior. Please let me know once you've addressed these and I'll have another look.

**Comment by davidsbailey** [comment]
LGTM

**Comment by joshlory** [comment]
I don't think we want this extra outline for the placeholder image:

placeholder-outline-bug

File: dashboard/public/apps-package/js/studio.js [hunk 5]
```diff
@@ -8202,24 +8205,25 @@ levels.js_hoc2015_event_two_items = {
...
-    [0, 0, 0, 0,  0, 0, 0, 0], 
-    [0, 0, 0, 1,  0, 0, 0, 0]],
+    [0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00],
+    [0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00],
+    [0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00],
+    [0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00],
+    [0x00, 0x00, 0x00, 0x10, 0x00, 0x00, 0x00, 0x00],
+    [0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00],
+    [0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00],
+    [0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00]],
```
Comment (joshlory) [inline dashboard/public/apps-package/js/studio.js:8215]: Why this change?

Comment (davidsbailey) [inline dashboard/public/apps-package/js/studio.js:8215]: Mehal accidentally pulled some other commits from staging into his branch. you can see from the merge commit that these changes had no effect.
